### ISSUE TITLE
Properties from Object.assign(module.exports) escape type-checking

### ISSUE THREAD

**Issue by nichtverstehen** [google/closure-compiler#2929]
I'm compiling the following with `--process_common_js_modules`:
```
// a.js
const c = require("./c");
console.log(c.foo);

// c.js
module.exports = {};
Object.assign(module.exports, {
  foo: 5,
});
```

and get an error:
```
a.js:11: WARNING - Property foo never defined on module$experimental$users$cyriln$js$c.default
console.log(c.foo);
              ^^^
```

even though the compiled output is functional:
```
// some $jscomp polyfills omitted
var module$experimental$users$cyriln$js$c$default = {};
Object.assign(module$experimental$users$cyriln$js$c$default, {foo:5});
console.log(module$experimental$users$cyriln$js$c$default.foo);
```

However, if the `Object.assign` is applied to a regular variable, typing works fine:
```
var c = {};
Object.assign(c, {
  foo: 5,
});

module.exports = c;
```
and generates exactly same code.

Something is not right about processing `module.exports`.

**Comment by ray007** [comment]
The `module.exports` assignment seems to work, but the type-checker doesn't seem to recognize the properties on the assigned object.
I've changed all my to code to `exports.prop = value-or-function` in order to avoid this problem.

**Comment by brad4d** [comment]
I don't believe the compiler currently any custom logic it would need to recognize `Object.assign` as adding properties to an object and updating type information accordingly.

**Comment by ray007** [comment]
> I don't believe the compiler currently any custom logic it would need to recognize Object.assign as adding properties to an object and updating type information accordingly.

That's what `@lends` is for, though that has its own problems with prototypes...
